src/MuTalk-Model/MTSubclassReplacementOperator.class.st

```diff
@@ -0,0 +1,93 @@
+Class {
+	#name : 'MTSubclassReplacementOperator',
+	#superclass : 'MTBlockBasedMutantOperator',
+	#instVars : [
+		'classesToReplace'
+	],
+	#category : 'MuTalk-Model-Operators',
+	#package : 'MuTalk-Model',
+	#tag : 'Operators'
+}
+
+{ #category : 'printing' }
+MTSubclassReplacementOperator >> description [
+
+	^ 'Replace a class reference by a reference to one of its subclasses'
+]
+
+{ #category : 'applying' }
+MTSubclassReplacementOperator >> expressionToReplace [
+
+	^ [ :aNode |
+	  aNode isVariable and: [
+		  (self class environment includesKey: aNode name) and: [
+			  | referencedClass shouldReplace |
+			  referencedClass := self class environment at: aNode name.
+			  shouldReplace := referencedClass isClass and: [
+				                   referencedClass subclasses notEmpty ].
+			  shouldReplace ifTrue: [ classesToReplace add: referencedClass ].
+			  shouldReplace ] ] ]
```

Comment:
I want that we discuss how to fix this. I know you just followed how others were done, but I don't like blocks used like this!